pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -652,6 +653,10 @@ func (s *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 			return errors.Errorf("MachineConfig managedIdentity is missing on machine %s", s.scope.Machine.Name)
 		}
 
+		if s.scope.Machine.Labels[machinev1.MachineClusterIDLabel] == "" {
```

Comment:
Is there any possibility that `s.scope.Machine.Labels` can be nil? We've tripped over this before where something trounces on `.Labels`.

Reply:
Good catch!!!. Resolved.